CMakeLists.txt

```diff
@@ -6,28 +6,58 @@ endif()
 
 cmake_minimum_required(VERSION 3.0)
 project(Luau LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
We shouldn't be setting global options, instead preferring the per-target specification like we do for other targets